src/main/resources/META-INF/jqassistant-rules/spring-injection.xml

```diff
@@ -67,6 +70,10 @@
                 method:Constructor
                 AND (type)-[:EXTENDS*0..]->(injectable)
               )
+              OR (
+                method.visibility = 'private'
+                AND (:Spring:BeanProducer)-[:INVOKES*]->(method)
```

Comment:
IMO, this weakens the constraint too much: It skips private methods as long as there is **any** path with **any** complexity from the producer to said method.

What we acutally want to allow are private methods, that are **exclusively** and **immediately** used by the producer. (If not immediately, then at least with limited complexity and only via other private methods).